src/_actions/TradeActions.js

```diff
@@ -171,6 +171,17 @@ export const updatePurchaseInfo = (index, fieldName, fieldValue) => ({
     fieldValue,
 });
 
+// TODO: this is arbritary, need discuss how we should handle it
+const trySubscribeUntilSuccess = c =>
+    LiveData.api
+        .subscribeToOpenContract(c)
+        .then(response => {
+            const contract = response.proposal_open_contract;
+            if (contract.validation_error && Object.keys(contract).length < 3) {
+                setTimeout(() => trySubscribeUntilSuccess(c), 300);
+            }
+        });
```

Comment:
Take note this would mean there's a chance, purchased contract will not be shown in our portfolio, unless user refresh

Should be edge case, reported by binary-tester